### PR TITLE
feat: show enrollment status in enrollments table

### PR DESCRIPTION
**Pull request by julianramirez2**
## Description

Now, in the enrollments, an additional piece of information will be displayed, which is the status. The status can be 'success,' 'error,' or 'no process,' indicating whether the last request was successful, encountered an error, or if no process was necessary (in the case of saving the enrollment locally).

## Testing instructions

Pull the branch locally and run the 'composer install' command to install all plugin dependencies and test. 

## Additional information

To filter and search for enrollments based on their status, it's as simple as typing the desired status into the search bar located at the top right corner of the enrollments.

## Checklist for Merge

- [X] Tested in a remote environment
- [ ] Updated documentation N/A
- [ ] Rebased master/main N/A
- [ ] Squashed commits N/A

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @julianramirez2!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md)     file for more information.     If you've signed an agreement in the past, you may need to re-sign.  See     [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/)     for details.

    
Once you've signed the CLA, please allow 1 business day for it to be processed.  After this time, you can re-run the     CLA check by adding a comment here that you have signed it.  If the problem persists, you can tag the `@openedx/cla-problems` team in a     comment on your PR for further assistance.

File: includes/model/class-openedx-woocommerce-plugin-log.php [hunk 1]
```diff
@@ -87,6 +87,9 @@ public function check_api_response( $response ) {
 			case 'not_api':
 				return 'Enrollment saved locally, action does not require an API call';
 
+			case 'no_process':
```
Comment (MaferMazu) [inline includes/model/class-openedx-woocommerce-plugin-log.php:90]: If we are going to create this case, do we need the default case? In what scenario does the plugin return 'API did not provide a response'?
Can we use a return for the case 'no_process' more descriptive? In the other case, we return a string, but in that specific case, we return a response[1]. Is it not better to return the message you defined, 'The enrollment process has been skipped.', here?

**Review by MaferMazu — COMMENTED** [review]
Thanks for this PR @julianramirez2, the feature is working as expected.
With this change you are going to be able to see the status of the last requests.
image
I have some questions only for improve the code, but the rest looks good to me.

**Comment by julianramirez2** [comment]
@andres-arango-aptitude can you check this, please?

**Review by MaferMazu — APPROVED** [review]
Thank @julianramirez2, for resolving the issues.

**Comment by openedx-webhooks** [comment]
@julianramirez2 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-wordpress-ecommerce&entry.867055334=https://github.com/openedx/openedx-wordpress-ecommerce/pull/56&entry.1484655318=https://github.com/julianramirez2&entry.752974735=2023-10-26+00:12&entry.1917517419=2023-11-08+23:21&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**